pkg/services/clustering/cluster_manager.go

```diff
@@ -96,16 +97,29 @@ func (cm *ClusterManager) clusterMgrTicker(ctx context.Context) error {
 	}()
 	cm.log.Info("clusterMgrTicker started")
 	ticksCounter := 0
+	cleanupCheck := false
 	for {
 		select {
 		case <-ctx.Done():
 			cm.log.Info("clusterMgrTicker Done")
 			return ctx.Err()
 		case <-cm.ticker.C: // ticks every second
 			if ticksCounter%60 == 0 {
-				cm.clusterNodeMgmt.CheckIn(cm.alertingState)
+				if cm.alertingState.run_type != m.CLN_ALERT_RUN_TYPE_CLEANUP {
+					if err := cm.clusterNodeMgmt.CheckIn(cm.alertingState, -1); err != nil {
+						cm.log.Error("Failed to checkin", "error", err.Error())
+					}
+				}
+			}
+			if ticksCounter%setting.ClusteringCleanupPeriod == 0 {
+				cleanupCheck = true
 			}
 			if ticksCounter%30 == 0 {
+				if cleanupCheck {
+					if cm.cleanupScheduler() {
+						cleanupCheck = false
```

Comment:
If the cleanup job is not completed in 30 seconds then it will trigger a new cleanupSchedular.